build error on RHEL8.2

```
  [ shim_syscalls.o ]
In file included from shim_syscalls.c:19:
../include/shim_table.h:745:66: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 int shim_timer_settime(timer_t timer_id, int flags, const struct __kernel_itimerspec* new_setting,
                                                                  ^~~~~~~~~~~~~~~~~~~
../include/shim_table.h:747:49: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 int shim_timer_gettime(timer_t timer_id, struct __kernel_itimerspec* setting);
                                                 ^~~~~~~~~~~~~~~~~~~
../include/shim_table.h:815:59: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 int shim_timerfd_settime(int ufd, int flags, const struct __kernel_itimerspec* utmr,
                                                           ^~~~~~~~~~~~~~~~~~~
../include/shim_table.h:817:42: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 int shim_timerfd_gettime(int ufd, struct __kernel_itimerspec* otmr);
                                          ^~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:768:41: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
                            const struct __kernel_itimerspec*, new_setting,
                                         ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:376:37: note: in definition of macro ‘SHIM_SYSCALL_PROTO_4’
     RTYPE shim_##NAME(T1 P1, T2 P2, T3 P3, T4 P4)
                                     ^~
shim_syscalls.c:767:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_settime, 4, int, timer_t, timer_id, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:376:11: error: conflicting types for ‘shim_timer_settime’
     RTYPE shim_##NAME(T1 P1, T2 P2, T3 P3, T4 P4)
           ^~~~~
../include/shim_internal.h:385:5: note: in expansion of macro ‘SHIM_SYSCALL_PROTO_4’
     SHIM_SYSCALL_PROTO_##n(name, __VA_ARGS__);                                     \
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:767:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_settime, 4, int, timer_t, timer_id, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:745:5: note: previous declaration of ‘shim_timer_settime’ was here
 int shim_timer_settime(timer_t timer_id, int flags, const struct __kernel_itimerspec* new_setting,
     ^~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:768:41: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
                            const struct __kernel_itimerspec*, new_setting,
                                         ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:186:38: note: in definition of macro ‘PROTO_ARGS_2’
 #define PROTO_ARGS_2(t, a, rest ...) t a, PROTO_ARGS_1(rest)
                                      ^
../include/shim_internal.h:188:43: note: in expansion of macro ‘PROTO_ARGS_3’
 #define PROTO_ARGS_4(t, a, rest ...) t a, PROTO_ARGS_3(rest)
                                           ^~~~~~~~~~~~
../include/shim_internal.h:211:20: note: in expansion of macro ‘PROTO_ARGS_4’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
                    ^~~~~~~~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:767:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_settime, 4, int, timer_t, timer_id, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:211:7: error: conflicting types for ‘shim_timer_settime’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
       ^~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:767:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_settime, 4, int, timer_t, timer_id, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:745:5: note: previous declaration of ‘shim_timer_settime’ was here
 int shim_timer_settime(timer_t timer_id, int flags, const struct __kernel_itimerspec* new_setting,
     ^~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:771:77: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 SHIM_SYSCALL_RETURN_ENOSYS(timer_gettime, 2, int, timer_t, timer_id, struct __kernel_itimerspec*,
                                                                             ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:370:30: note: in definition of macro ‘SHIM_SYSCALL_PROTO_2’
     RTYPE shim_##NAME(T1 P1, T2 P2)
                              ^~
shim_syscalls.c:771:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_gettime, 2, int, timer_t, timer_id, struct __kernel_itimerspec*,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:370:11: error: conflicting types for ‘shim_timer_gettime’
     RTYPE shim_##NAME(T1 P1, T2 P2)
           ^~~~~
../include/shim_internal.h:385:5: note: in expansion of macro ‘SHIM_SYSCALL_PROTO_2’
     SHIM_SYSCALL_PROTO_##n(name, __VA_ARGS__);                                     \
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:771:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_gettime, 2, int, timer_t, timer_id, struct __kernel_itimerspec*,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:747:5: note: previous declaration of ‘shim_timer_gettime’ was here
 int shim_timer_gettime(timer_t timer_id, struct __kernel_itimerspec* setting);
     ^~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:771:77: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 SHIM_SYSCALL_RETURN_ENOSYS(timer_gettime, 2, int, timer_t, timer_id, struct __kernel_itimerspec*,
                                                                             ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:185:28: note: in definition of macro ‘PROTO_ARGS_1’
 #define PROTO_ARGS_1(t, a) t a
                            ^
../include/shim_internal.h:211:20: note: in expansion of macro ‘PROTO_ARGS_2’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
                    ^~~~~~~~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:771:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_gettime, 2, int, timer_t, timer_id, struct __kernel_itimerspec*,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:211:7: error: conflicting types for ‘shim_timer_gettime’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
       ^~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:771:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timer_gettime, 2, int, timer_t, timer_id, struct __kernel_itimerspec*,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:747:5: note: previous declaration of ‘shim_timer_gettime’ was here
 int shim_timer_gettime(timer_t timer_id, struct __kernel_itimerspec* setting);
     ^~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:964:41: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
                            const struct __kernel_itimerspec*, utmr, struct __kernel_itimerspec*,
                                         ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:376:37: note: in definition of macro ‘SHIM_SYSCALL_PROTO_4’
     RTYPE shim_##NAME(T1 P1, T2 P2, T3 P3, T4 P4)
                                     ^~
shim_syscalls.c:963:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_settime, 4, int, int, ufd, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:376:11: error: conflicting types for ‘shim_timerfd_settime’
     RTYPE shim_##NAME(T1 P1, T2 P2, T3 P3, T4 P4)
           ^~~~~
../include/shim_internal.h:385:5: note: in expansion of macro ‘SHIM_SYSCALL_PROTO_4’
     SHIM_SYSCALL_PROTO_##n(name, __VA_ARGS__);                                     \
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:963:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_settime, 4, int, int, ufd, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:815:5: note: previous declaration of ‘shim_timerfd_settime’ was here
 int shim_timerfd_settime(int ufd, int flags, const struct __kernel_itimerspec* utmr,
     ^~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:964:41: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
                            const struct __kernel_itimerspec*, utmr, struct __kernel_itimerspec*,
                                         ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:186:38: note: in definition of macro ‘PROTO_ARGS_2’
 #define PROTO_ARGS_2(t, a, rest ...) t a, PROTO_ARGS_1(rest)
                                      ^
../include/shim_internal.h:188:43: note: in expansion of macro ‘PROTO_ARGS_3’
 #define PROTO_ARGS_4(t, a, rest ...) t a, PROTO_ARGS_3(rest)
                                           ^~~~~~~~~~~~
../include/shim_internal.h:211:20: note: in expansion of macro ‘PROTO_ARGS_4’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
                    ^~~~~~~~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:963:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_settime, 4, int, int, ufd, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:211:7: error: conflicting types for ‘shim_timerfd_settime’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
       ^~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:963:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_settime, 4, int, int, ufd, int, flags,
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:815:5: note: previous declaration of ‘shim_timerfd_settime’ was here
 int shim_timerfd_settime(int ufd, int flags, const struct __kernel_itimerspec* utmr,
     ^~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:967:70: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_gettime, 2, int, int, ufd, struct __kernel_itimerspec*, otmr)
                                                                      ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:370:30: note: in definition of macro ‘SHIM_SYSCALL_PROTO_2’
     RTYPE shim_##NAME(T1 P1, T2 P2)
                              ^~
shim_syscalls.c:967:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_gettime, 2, int, int, ufd, struct __kernel_itimerspec*, otmr)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:370:11: error: conflicting types for ‘shim_timerfd_gettime’
     RTYPE shim_##NAME(T1 P1, T2 P2)
           ^~~~~
../include/shim_internal.h:385:5: note: in expansion of macro ‘SHIM_SYSCALL_PROTO_2’
     SHIM_SYSCALL_PROTO_##n(name, __VA_ARGS__);                                     \
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:967:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_gettime, 2, int, int, ufd, struct __kernel_itimerspec*, otmr)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:817:5: note: previous declaration of ‘shim_timerfd_gettime’ was here
 int shim_timerfd_gettime(int ufd, struct __kernel_itimerspec* otmr);
     ^~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:18:
shim_syscalls.c:967:70: warning: ‘struct __kernel_itimerspec’ declared inside parameter list will not be visible outside of this definition or declaration
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_gettime, 2, int, int, ufd, struct __kernel_itimerspec*, otmr)
                                                                      ^~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:185:28: note: in definition of macro ‘PROTO_ARGS_1’
 #define PROTO_ARGS_1(t, a) t a
                            ^
../include/shim_internal.h:211:20: note: in expansion of macro ‘PROTO_ARGS_2’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
                    ^~~~~~~~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:967:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_gettime, 2, int, int, ufd, struct __kernel_itimerspec*, otmr)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/shim_internal.h:211:7: error: conflicting types for ‘shim_timerfd_gettime’
     r shim_##name (PROTO_ARGS_##n (args)) {                         \
       ^~~~~
../include/shim_internal.h:391:5: note: in expansion of macro ‘EXPORT_SHIM_SYSCALL’
     EXPORT_SHIM_SYSCALL(name, n, __VA_ARGS__)
     ^~~~~~~~~~~~~~~~~~~
shim_syscalls.c:967:1: note: in expansion of macro ‘SHIM_SYSCALL_RETURN_ENOSYS’
 SHIM_SYSCALL_RETURN_ENOSYS(timerfd_gettime, 2, int, int, ufd, struct __kernel_itimerspec*, otmr)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from shim_syscalls.c:19:
../include/shim_table.h:817:5: note: previous declaration of ‘shim_timerfd_gettime’ was here
 int shim_timerfd_gettime(int ufd, struct __kernel_itimerspec* otmr);
     ^~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:153: shim_syscalls.o] Error 1

```

--- a/LibOS/shim/include/shim_types.h
+++ b/LibOS/shim/include/shim_types.h
@@ -56,7 +56,9 @@ struct __kernel_timespec {
     __kernel_time_t tv_sec;         /* seconds */
     long            tv_nsec;        /* nanoseconds */
 };
+#endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 0)
 struct __kernel_itimerspec {
     struct __kernel_timespec it_interval;    /* timer period */
     struct __kernel_timespec it_value;       /* timer expiration */

